xapian-core/include/xapian/weight.h

```diff
@@ -29,6 +29,10 @@
 #include <xapian/types.h>
 #include <xapian/visibility.h>
 
+#include "registry.h"
+
+#include "../common/stringutils.h"
```

Comment:
You can't include headers internal to the library from a public API header (but I can't see why you need to).